tests/models/encodec/test_modeling_encodec.py

```diff
@@ -375,17 +374,13 @@ def check_equivalence(model, tuple_inputs, dict_inputs, additional_kwargs={}):
                 tuple_output = model(**tuple_inputs, return_dict=False, **additional_kwargs)
                 dict_output = model(**dict_inputs, return_dict=True, **additional_kwargs)
 
+                self.assertTrue(isinstance(tuple_output, tuple))
+                self.assertTrue(isinstance(dict_output, dict))
+
                 def recursive_check(tuple_object, dict_object):
-                    if isinstance(tuple_object, (List, Tuple)):
-                        for tuple_iterable_value, dict_iterable_value in zip(tuple_object, dict_object):
-                            recursive_check(tuple_iterable_value, dict_iterable_value)
-                    elif isinstance(tuple_object, Dict):
```

Comment:
Even if we assert that `tuple_object` is a tuple on L377, as this is a recursive function, isn't is still possible that the values in tuple_object i.e. `tuple_iterable_value` are a dict or None?

Reply:
The values in the tuple_object should be `tensor.Tensor` only.

To check the types, I added a print statement in the recursive function:

```py
def recursive_check(tuple_object, dict_object):
    print(f"[DEBUG]: {type(tuple_object)=}, {type(dict_object)=}")
    ...
```

and I got this:

```py
[DEBUG]: type(tuple_object)=<class 'torch.Tensor'>, type(dict_object)=<class 'torch.Tensor'>
[DEBUG]: type(tuple_object)=<class 'torch.Tensor'>, type(dict_object)=<class 'torch.Tensor'>
```